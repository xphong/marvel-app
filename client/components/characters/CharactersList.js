import PropTypes from 'prop-types';
import React, { Component } from 'react';

import Character from './Character';

export default class CharactersList extends Component {
  constructor(props) {
    super(props);
  }

  renderCharacter(character) {
    return <Character key={character.name} character={character} />;
  }

  renderServerError() {
    return (
      <div className="ui message red">
        Server Error
      </div>
    );
  }

  renderCharactersFound() {
    return (
      <div className="ui message orange">
        No characters found
      </div>
    );
  }

  render() {
    const { characters } = this.props;

    if (characters.error) {
      return this.renderServerError();
    }

    if (!characters.data.length && characters.isFetched) {
      return this.renderCharactersFound();
    }

    return (
      <div className="ui characters-list cards">
        {characters.data.map(this.renderCharacter)}
      </div>
    );
  }
}

CharactersList.propTypes = {
  characters: PropTypes.object.isRequired
};
