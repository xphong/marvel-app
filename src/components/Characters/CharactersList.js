import React, { Component, PropTypes } from 'react';

import Character from './Character';

export default class CharactersList extends Component {
  constructor(props){
    super(props);
  }

  renderCharacter(character) {
    return <Character key={character.name} character={character} />;
  }

  render() {
    const {data} = this.props.characters;

    return (
      <div className='ui characters-list cards'>
        {data.map(this.renderCharacter)}
      </div>
    );
  }
}

CharactersList.propTypes = {
  characters: PropTypes.object.isRequired
};
