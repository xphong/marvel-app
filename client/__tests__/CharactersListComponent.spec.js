import React from 'react';
import ReactDOM from 'react-dom';

import CharactersList from '../components/characters/CharactersList';

describe('CharactersList Component', () => {
  it('should render CharactersList', () => {
    const props = {
      characters: {
          data: [
              {
                name: 'Hawkeye',
                description: 'poop'
              }
          ]
      }
    };

    const wrapper = shallow(<CharactersList {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
