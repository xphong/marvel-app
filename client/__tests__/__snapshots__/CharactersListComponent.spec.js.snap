// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharactersList Component should render CharactersList 1`] = `
<div
  className="ui characters-list cards"
>
  <Character
    character={
      Object {
        "description": "poop",
        "name": "Hawkeye",
      }
    }
    key="Hawkeye"
  />
</div>
`;
